### PR TITLE
Add host and sitemap to robots.txt

### PR DESCRIPTION
**Pull request by zchase**
Fixes: https://github.com/pulumi/home/issues/1838

### CODE REVIEW

File: themes/default/layouts/robots.txt [hunk 1]
```diff
@@ -5,3 +5,6 @@ Disallow: *
 
 # Disallow certain pages that are preceived as soft-404s.
 Disallow: /docs/search/$
+
+Sitemap: https://www.pulumi.com/sitemap.xml
+Host: https://www.pulumi.com
```
Comment (cnunciato) [inline themes/default/layouts/robots.txt:10]: Looking at https://developers.google.com/search/docs/advanced/robots/robots_txt#syntax, I don't see `Host` listed as a property Google supports, and I don't see it listed in the spec at https://datatracker.ietf.org/doc/html/draft-rep-wg-topic-00 (assuming I'm not looking in an outdated place). @Brant-Soler is there guidance that explains this prop in more detail?

Comment (cnunciato) [reply themes/default/layouts/robots.txt:10]: Looks like based on [this recommendation](https://yoast.com/ultimate-guide-robots-txt/#host-directive), we should probably omit the protocol bit:

```suggestion
Host: www.pulumi.com
```

**Review by cnunciato — APPROVED** [review]
One recommendation to tweak the `Host` directive, otherwise 👍!

**Comment by Brant-Soler** [comment]
Yes, absolutely. Make it so...

**Pull request merged**